File: app/src/main/java/com/ababilx/routinescrapper/ui/student/components/DecorBlobs.kt
package com.ababilx.routinescrapper.ui.student.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.unit.dp
import com.ababilx.routinescrapper.ui.theme.Lavender
import com.ababilx.routinescrapper.ui.theme.Mint
import com.ababilx.routinescrapper.ui.theme.Peach

@Composable
fun DecorBlobs(modifier: Modifier = Modifier) {
    Canvas(modifier.fillMaxSize()) {
        val mint = Mint.copy(alpha = 0.42f)
        val peach = Peach.copy(alpha = 0.38f)
        val lavender = Lavender.copy(alpha = 0.32f)
        drawCircle(mint, 160.dp.toPx(), Offset(size.width * 0.92f, 40.dp.toPx()))
        drawCircle(lavender, 130.dp.toPx(), Offset(-10.dp.toPx(), size.height * 0.42f))
        drawCircle(peach, 110.dp.toPx(), Offset(size.width * 0.88f, size.height * 0.78f))
    }
}

File: lib/ui/components/decor_blobs.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';

class DecorBlobs extends StatelessWidget {
  const DecorBlobs({super.key});

  @override
  Widget build(BuildContext context) {
    return const CustomPaint(painter: _BlobPainter(), child: SizedBox.expand());
  }
}

class _BlobPainter extends CustomPainter {
  const _BlobPainter();

  @override
  void paint(Canvas canvas, Size size) {
    canvas.drawCircle(
      Offset(size.width * 0.92, 40),
      160,
      Paint()..color = mint.withValues(alpha: 0.42),
    );
    canvas.drawCircle(
      Offset(-10, size.height * 0.42),
      130,
      Paint()..color = lavender.withValues(alpha: 0.32),
    );
    canvas.drawCircle(
      Offset(size.width * 0.88, size.height * 0.78),
      110,
      Paint()..color = peach.withValues(alpha: 0.38),
    );
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) => false;
}
